addons/point_of_sale/static/tests/tours/helpers/PartnerListTourMethods.js

```diff
@@ -13,3 +13,73 @@ export function clickPartnerDetails(name){
         trigger: `.partner-list tr:contains("${name}") button.edit-partner-button`,
     }
 }
+
+export function checkContactValues(name, address = "", phone = "", mobile = "", email = "") {
+    const steps = [
+        {
+            content: `Check partner "${name}" from partner list screen`,
+            trigger: `.partner-list td:contains("${name}")`,
+            run: () => {},
+        },
+        {
+            content: `Check address "${address}" for partner "${name}"`,
+            trigger: `.partner-list tr:contains("${name}") .partner-line-adress:contains("${address}")`,
+            run: () => {},
+        },
+    ];
+
+    if(phone) {
+        steps.push({
+            content: `Check phone number "${phone}" for partner "${name}"`,
+            trigger: `.partner-list tr:contains("${name}") .partner-line-email:contains("${phone}")`,
+            run: () => {},
+        });
+    }
+
+    if(mobile) {
+        steps.push({
+            content: `Check mobile number "${mobile}" for partner "${name}"`,
+            trigger: `.partner-list tr:contains("${name}") .partner-line-email:contains("${mobile}")`,
+            run: () => {},
+        });
+    }
+
+    if(email) {
+        steps.push({
+            content: `Check email address "${email}" for partner "${name}"`,
+            trigger: `.partner-list tr:contains("${name}") .partner-line-email .email-field:contains("${email}")`,
+            run: () => {},
+        });
+    }
+
+    return steps;
+}
+
+export function searchCustomer(val) {
+    return [
+        {
+            content: `Click search field`,
+            trigger: `.fa-search.undefined`,
+        },
+        {
+            content: `Search customer with "${val}"`,
+            trigger: `.input-group input`,
+            run: `text ${val}`,
+        },
+        {
+            content: `Check "${val}" is shown`,
+            trigger: `.partner-list tr:nth-child(1):contains("${val}")`,
+            run: () => {},
+        },
+
+        // Clear the search otherwise the checks are going too fast for the list to change
+        {
+            trigger: `.input-group input`,
+            run: `text Addison`,
+        },
```

Comment:
Here you select Addison to clear your search. However you haven't explicitly created a partner with the name Addison nor does it exist in the class variables if I'm not mistaken. So it means your test is relying on demo data. When tests rely on demo data we usually add `if not loaded_demo_data(self.env)` in the python test to avoid an error when trying to run the test without demo data.

Instead of relying on demo data you could just create another partner called Addison.

Another solution to avoid demo data and relying on a second partner could be to make this function do the following steps: Open customer list, **search steps**, close customer list. This will also reset the search field.